frappe/core/doctype/file/file.py

```diff
@@ -437,15 +441,37 @@ def get_content(self) -> bytes:
 			self.validate_file_url()
 		file_path = self.get_full_path()
 
+		if self.get_full_path().startswith(URL_PREFIXES):
+			if self.get_full_path().startswith("https://" + frappe.local.site + "/api/method/storage_integration"):
+				import storage_integration.controller as s3
```

Comment:
This should be handled in the app instead. 

@rutwikhdev ?

Reply:
> This should be handled in the app instead.
> 
> @rutwikhdev ?

@DrZoidberg09 
This could be also done by overriding before_insert. But you'll have to do this on your own fork. Don't think downloading should be the default behaviour for remote files even in the app.